src/main/java/net/snowflake/client/core/StmtUtil.java

```diff
@@ -105,6 +105,8 @@ static class StmtInput {
 
     QueryContextDTO queryContextDTO;
 
+    Map<String, String> additionalHttpHeaders;
```

Comment:
I see this map in SFLogin also. Can we abstract out and put at one place?

Reply:
I can make an abstract superclass, sure. I mainly didn't want to make too big of an incision.

Reply:
Added an abstract superclass. In the interest of not broadening the scope and making a larger PR, I have not moved more properties. I can do that in a followup PR if needed for this one.